lib/smartcar/base.rb

```diff
@@ -23,11 +24,15 @@ class InvalidParameterValue < StandardError; end
       # @param data [Hash] request body if needed.
       #
       # @return [Hash] The response Json parsed as a hash.
-      define_method verb do |path, data = nil|
+      define_method verb do |path, data = nil, headers = {}|
         response = service.send(verb) do |request|
-          request.headers['Authorization'] = auth_type == BASIC ? "Basic #{token}" : "Bearer #{token}"
-          request.headers['sc-unit-system'] = unit_system if unit_system
-          request.headers['Content-Type'] = 'application/json'
+          request_headers = {}
+          request_headers['Authorization'] = auth_type == BASIC ? "Basic #{token}" : "Bearer #{token}"
+          request_headers['sc-unit-system'] = unit_system if unit_system
+          request_headers['Content-Type'] = 'application/json'
+          request_headers['User-Agent'] =
+            "Smartcar/#{version} (#{RbConfig::CONFIG['host_os']}; #{RbConfig::CONFIG['arch']}) Ruby v#{RUBY_VERSION}"
```

Comment:
This is the API version (v1 vs v2) we need the SDKs current verson here in the `VERSION` constant